### PR TITLE
fix open setting

### PR DESCRIPTION
**Pull request by chung-lai**
There is a timing issue between when the deep link is handled in notification and the app state. This should be a safe fix but we should look for a better solution to make sure the app state is set up properly before handling the notification deeplink.

### How was this tested?
- [x] I tested this manually
- [ ] I added automated tests
- [ ] Existing automated tests are enough to cover my changes

https://user-images.githubusercontent.com/87332578/173922454-72fbdf19-2bd7-4c17-8c53-396461f5eb22.mov

### CODE REVIEW

File: Client/Application/NavigationRouter.swift [hunk 1]
```diff
@@ -274,7 +274,9 @@ enum NavigationPath {
         guard let page = page else {
             return
         }
-        bvc.openSettings(openPage: page)
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
```
Comment (jedireza) [inline Client/Application/NavigationRouter.swift:277]: Have we tried shorter than 1 second?

Comment (chung-lai) [reply Client/Application/NavigationRouter.swift:277]: i tried 500ms and it seem to work fine. I'll shorten it

File: Client/Application/NavigationRouter.swift [hunk 2]
```diff
@@ -274,7 +274,9 @@ enum NavigationPath {
         guard let page = page else {
             return
         }
-        bvc.openSettings(openPage: page)
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```
Comment (jedireza) [inline Client/Application/NavigationRouter.swift:277]:
```suggestion
        // TODO: look for a better solution to make sure the app state is set up properly before handling the notification deep link
        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```

**Review by jedireza — APPROVED** [review]
Cool. Thanks for adding the video. I suggest adding a TODO comment and filing a followup issue to track.

**Pull request merged**